### ISSUE TITLE
[BUG] Firefox locator click behaves differently than other browsers

### ISSUE THREAD

**Issue by sean-perkins** [microsoft/playwright#20168]
**Context:**
- Playwright Version: v1.29.1
- Operating System: Mac
- Node.js version: v16.15.1
- Browser: Firefox

**Code Snippet**

- https://github.com/ionic-team/ionic-framework

Reproduction steps:
- Clone repository
- `npm intall`
- `cd core`
- `npm install`
- Remove the `test.skip` for firefox here: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/popover/test/dismissOnSelect/popover.e2e.ts#L28-L31
- `npm run build` (from `core` directory)
- `npx playwright test src/components/popover/test/dismissOnSelect/`
- Observe: Tests are failing only on Firefox

Example trace: [f9144bf946ba44905f0e2bd635685a58adc62f5b.zip](https://github.com/microsoft/playwright/files/10438285/f9144bf946ba44905f0e2bd635685a58adc62f5b.zip)

**Describe the bug**

We have encountered a difference in behavior between testing against Firefox and other browser vendors (Safari, Chrome) with using the `locator.click` API. 

We have a nested popover (dropdown) test that validates that the parent popover remains visible when the child popover is activated: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/popover/test/dismissOnSelect/popover.e2e.ts#L37-L40

The tests will pass consistently for Chrome and Safari, but fails on Firefox (we are currently skipping it as a result)

In the trace we can see that the parent popover is dismissing only in Firefox and the child popover remains visible. This behavior is however not reproducible when manually testing in the browser. 

If I change the test code from:
```ts

await clickTrigger.click();
```

to:
```ts
await clickTrigger.evaluate((el: HTMLButtonElement) => el.click());
```

All browsers will fail consistently with the same end behavior as Firefox is showing for the previous usage. It appears that the Firefox emulation for clicking the locator has different behavior than Chrome/Safari. 

Our component does have some extra overhead with `requestAnimationFrame` and delegating focus, but with the UI working as intended in manual testing; I'm unsure if this is an Ionic or Playwright issue, with my limited knowledge of how Playwright works internally.

**Comment by sean-perkins** [comment]
I've debugged this further and it seems that stopping the event propagation is contributing to this behavior for Firefox: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/popover/utils.ts#L135

However it is needed in the implementation, but has a negative impact when testing with playwright.

**Comment by aslushnikov** [comment]
I can repro on 1.29; this is fixed on `@playwright/test@1.31.0-alpha-feb-7-2023`. Will be fixed as part of 1.31.
